pkg/controllers/dynakube/extension/eec/reconciler.go

```diff
@@ -70,3 +86,100 @@ func (r *reconciler) Reconcile(ctx context.Context) error {
 
 	return r.createOrUpdateStatefulset(ctx)
 }
+
+func (r *reconciler) reconcileTlsSecret(ctx context.Context) error {
+	query := k8ssecret.Query(r.client, r.client, log)
+
+	if r.dk.GetExtensionsTlsRefName() != "" {
+		return query.DeleteForNamespaces(ctx, getSelfSignedTlsSecretName(r.dk.Name), []string{r.dk.Namespace})
+	}
+
+	secret, err := query.Get(ctx, types.NamespacedName{
+		Name:      getSelfSignedTlsSecretName(r.dk.Name),
+		Namespace: r.dk.Namespace,
+	})
+
+	if err != nil && !k8serrors.IsNotFound(err) {
+		return err
+	}
+
+	if k8serrors.IsNotFound(err) {
+		err = r.createOrUpdateTlsSecret(ctx)
+		if err != nil {
+			return err
+		}
+
+		return nil
+	}
+
+	err = r.reconcileTlsSecretExpiration(ctx, secret)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (r *reconciler) createOrUpdateTlsSecret(ctx context.Context) error {
```

Comment:
can this whole implementation be made more generic so it can be reused for other components as well?